src/BaseFactory.php

```diff
@@ -40,37 +40,35 @@ protected function build(array $extra = [], string $creationType = 'create')
         $modelData = $this->prepareModelData($creationType, $this->getDefaults($this->faker));
         $model = $this->modelClass::$creationType(array_merge($modelData, $this->overwriteDefaults, $extra));
 
-        if ($this->relatedModels->isEmpty()) {
+        if ($this->relatedModelFactories->isEmpty()) {
             return $model;
         }
 
+        $relatedModels = $this->relatedModelFactories->map(fn($factory) => $factory->make());
```

Comment:
here we could also use the high order version, right? `->map->make();`

Reply:
Yup, I always forget about those. Refactored BaseFactory and MultiFactoryCollection to use these